src/components/Importer.js

```diff
@@ -0,0 +1,68 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+
+import toml from 'toml';
+
+import {mix, clone} from '../util';
+
+class Importer extends PureComponent {
+
+  static propTypes = {
+    defaults: PropTypes.object.isRequired,
+    onChange: PropTypes.func.isRequired,
+    onError: PropTypes.func.isRequired
+  };
+
+  handleFileChange (ev) {
+    const file = ev.target.files[0];
+    const domTarget = ev.target;
+    domTarget.value = '';
+
+    if (!window.confirm('Do you want to overwrite current config?')) {
+      return;
+    }
+
+    if (file.name.match(/.*\.toml$/)) {
+      var reader = new window.FileReader();
+
+      reader.onload = _ => {
+        let importedData;
+
+        try {
+          importedData = toml.parse(reader.result);
+        } catch (err) {
+          this.props.onError(`Couldn't parse configuration file. ${err.toString()}`);
+          return;
+        }
+
+        const data = mix(clone(this.props.defaults), importedData);
+        domTarget.blur();
+        this.props.onChange(data);
+      };
+
+      reader.readAsText(file);
+    } else {
+      this.props.onError('Please import a .toml configuration file.');
+    }
+  }
+
+  render () {
+    return (
+      <button
+        className='mdl-button mdl-button--icon mdl-js-button mdl-js-ripple-efect'
+        onClick={() => this.refs.fileInput.click()}>
+
+        <input
+          style={{display: 'none'}}
```

Comment:
For performance reasons all callbacks and `style` should not be re-computed on every render.

Reply:
Cool! Didn't know about this. But because it's a PureComponent the render method gets called very rarely; and I read that doing `.bind(this)` also reallocates a new variable ; ended up using https://reactjs.org/docs/faq-functions.html#bind-in-constructor-es2015 although it might be overkill.

Reply:
Actually a nice trick is to use lambdas for that, so instead of:
```
class X {
 constructor() {
   this.something = this.something.bind(this);
 }
 something () { console.log(this) }
}
```
you do
```
class X {
  something = () => console.log(this);
}
```
and `bind` is not needed.

Reply:
ah, nice! :+1: